lib/iiif_print/split_pdfs/pdf_child_works_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module IiifPrint
+  module SplitPdfs
+    ## Encapsulates logic for cleanup when the PDF is destroyed after pdf splitting into child works
+    class PdfChildWorksService
+      def self.create_pdf_child_works_for(file_set:, user:)
+        locations = pdfs_only_for([Hyrax::WorkingDirectory.find_or_retrieve(file.id, file_set.id)])
+        return if locations.empty?
+        work = file_set.parent
```

Comment:
```suggestion
        work = IiifPrint.parent_for(file_set)
```

We've seen cases where the `file_set.parent` is showing odd behavior.